workflows/cloudify_system_workflows/tests/test_plugins.py

```diff
@@ -0,0 +1,124 @@
+########
+# Copyright (c) 2019 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import unittest
+from functools import wraps
+
+from mock import patch, MagicMock, PropertyMock, call
+
+from cloudify import decorators
+from cloudify.models_states import ExecutionState
+
+from cloudify_system_workflows.plugins import update as update_func
+
+
+def mock_workflow_decorator(f=None, *_, **__):
+    if not f:
+        return lambda _f, *_, **__: mock_workflow_decorator(_f, *_, **__)
+
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        return f(*args, **kwargs)
+
+    return wrapper
+
+
+decorators.workflow = mock_workflow_decorator
+
+
+class TestPluginsUpdate(unittest.TestCase):
+    @patch('cloudify_system_workflows.plugins.get_rest_client')
+    @patch('cloudify_system_workflows.plugins.wait_for')
+    def test_stops_updating_when_one_update_fails(self,
+                                                  _,
+                                                  mock_get_rest_client):
+        def update_with_existing_blueprint_mock(deployment_id, *_, **__):
+            return PropertyMock(execution_id=deployment_id)
+
+        def returns_failed_execution_only_if_is_stop_at(execution_id):
+            if execution_id == stop_at:
+                return PropertyMock(status=test_params['curr_exec_status'])
+            return PropertyMock(status=ExecutionState.TERMINATED)
+
+        def _assert_update_func_raises():
+            with self.assertRaisesRegexp(
+                    RuntimeError,
+                    "Deployment update of deployment {0} with execution ID {0}"
```

Comment:
there is two {0}

Reply:
Yeah, the deployment ID and execution ID are the same